tests/unit/caches/test_directory.py

```diff
@@ -113,16 +122,7 @@ def test_key_lock_creates(self):
 
     @staticmethod
     def try_lock_from_another_process(cache_dir, key):
-        def run(cache_dir, key):
-            cache = Directory(path=cache_dir)
-            try:
-                cache.lock_key(key, timeout=0)
-                cache.unlock_key(key)
-            except:
-                sys.exit(1)
-            sys.exit(0)
-
-        process = multiprocessing.Process(target=run, args=(cache_dir, key))
+        process = multiprocessing.Process(target=try_cache_key_lock, args=(cache_dir, key,))
```

Comment:
Of course, multiprocessing on Windows behaves pretty differently... The target has to be a top level method. I believe that's because it's a requirement of being picklable.